presto-main/src/main/java/com/facebook/presto/operator/aggregation/LongMinMaxStateFactory.java

```diff
@@ -0,0 +1,186 @@
+    {
+        return new SingleLongMinMaxState(type, methodHandle);
+    }
+
+    @Override
+    public Class<? extends LongMinMaxState> getSingleStateClass()
+    {
+        return SingleLongMinMaxState.class;
+    }
+
+    @Override
+    public LongMinMaxState createGroupedState()
+    {
+        return new GroupLongMinMaxState(type, methodHandle);
+    }
+
+    @Override
+    public Class<? extends LongMinMaxState> getGroupedStateClass()
+    {
+        return GroupLongMinMaxState.class;
+    }
+
+    public static class GroupLongMinMaxState
+            extends AbstractGroupedAccumulatorState
+            implements LongMinMaxState
+    {
+        private final Type type;
+        private final MethodHandle methodHandle;
+        private final LongBigArray values = new LongBigArray();
+        private final BooleanBigArray nulls = new BooleanBigArray(true);
+
+        public GroupLongMinMaxState(Type type, MethodHandle methodHandle)
+        {
+            this.type = type;
+            this.methodHandle = methodHandle;
+        }
+
+        @Override
+        public void ensureCapacity(long size)
+        {
+            values.ensureCapacity(size);
+            nulls.ensureCapacity(size);
+        }
+
+        @Override
+        public long getEstimatedSize()
+        {
+            return values.sizeOf() + nulls.sizeOf();
+        }
+
+        @Override
+        public Type getType()
+        {
+            return type;
+        }
+
+        @Override
+        public MethodHandle getMethodHandle()
+        {
+            return methodHandle;
+        }
+
+        @Override
+        public long getLong()
+        {
+            return values.get(getGroupId());
+        }
+
+        @Override
+        public void setLong(long value)
+        {
+            values.set(getGroupId(), value);
+        }
+
+        @Override
+        public boolean isNull()
+        {
+            return nulls.get(getGroupId());
+        }
+
+        @Override
+        public void setNull(boolean value)
+        {
+            nulls.set(getGroupId(), value);
+        }
+    }
+
+    public static class SingleLongMinMaxState
+            implements LongMinMaxState
+    {
+        private final Type type;
+        private final MethodHandle methodHandle;
+        private long value;
+        private boolean isNull;
+        private static final int estimatedSize = ClassLayout.parseClass(SingleLongMinMaxState.class).instanceSize();
+
+        public SingleLongMinMaxState(Type type, MethodHandle methodHandle)
+        {
+            this.type = type;
+            this.methodHandle = methodHandle;
+            this.value = 0;
+            this.isNull = true;
+        }
+
+        @Override
+        public long getEstimatedSize()
+        {
+            return estimatedSize;
+        }
+
+        @Override
+        public Type getType()
+        {
+            return type;
+        }
+
+        @Override
+        public MethodHandle getMethodHandle()
```

Comment:
Oh ya, I hadn't thought of binding a MethodHandle into another MethodHandle like that. I think we can actually bind both the MethodHandle and the Type in the specialize function, similar to how `ArrayEqualOperator` works. That would save a lot of code, since then I don't think we need any of these state objects.